libs/utils/utils.js

```diff
@@ -863,6 +904,84 @@ function decorateMeta() {
   });
 }
 
+function decorateDocumentExtras(config) {
+  decorateMeta();
+  decorateHeader();
+  decorateFooterPromo(config);
+
+  import('./samplerum.js').then(({ addRumListeners }) => {
+    addRumListeners();
+  });
+}
+
+async function documentPostSectionLoading(config) {
+  const georouting = getMetadata('georouting') || config.geoRouting;
+  if (georouting === 'on') {
+    // eslint-disable-next-line import/no-cycle
+    const { default: loadGeoRouting } = await import('../features/georoutingv2/georoutingv2.js');
+    loadGeoRouting(config, createTag, getMetadata, loadBlock, loadStyle);
+  }
+  const appendage = getMetadata('title-append');
+  if (appendage) {
+    import('../features/title-append/title-append.js').then((module) => module.default(appendage));
+  }
+  if (getMetadata('seotech-structured-data') === 'on' || getMetadata('seotech-video-url')) {
+    import('../features/seotech/seotech.js').then((module) => module.default(
+      { locationUrl: window.location.href, getMetadata, createTag, getConfig },
+    ));
+  }
+  const richResults = getMetadata('richresults');
+  if (richResults) {
+    const { default: addRichResults } = await import('../features/richresults.js');
+    addRichResults(richResults, { createTag, getMetadata });
+  }
+  loadFooter();
+  const { default: loadFavIcon } = await import('./favicon.js');
+  loadFavIcon(createTag, getConfig(), getMetadata);
+  if (config.experiment?.selectedVariant?.scripts?.length) {
+    config.experiment.selectedVariant.scripts.forEach((script) => loadScript(script));
+  }
+  initSidekick();
+
+  const { default: delayed } = await import('../scripts/delayed.js');
+  delayed([getConfig, getMetadata, loadScript, loadStyle, loadIms]);
+}
+
+async function processSection(section, config, isDoc) {
+  const inlineFrags = [...section.el.querySelectorAll('a[href*="#_inline"]')];
```

Comment:
could we not return the inlineFrags as part of the return object in `decorateSection` so we don't need to select them here again / hit the DOM?

Reply:
How it's written now, decorateSection adds `#_inline` to all fragment links inside of blocks (excluding DO_NOT_INLINE blocks) but does not look for `#_inline` fragments outside of blocks.

You're right though that we could have `decorateLinks` return any `#_inline` links, though we'd also have to check that it's a fragment path.

I'll give it a try.

Reply:
Did some perf testing, the selector takes < 1ms to run, so I think I'd rather leave as is instead of having logic in `decorateLinks` and `decorateSection` to find and pass around the inline frags.